### PR TITLE
Limit the size of the image cache

### LINKED ISSUE #1009: image_cache grows unbounded
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/services/image_cache.dart#L38

@stevenroose reports "my app refreshes an image every minute ... and I noticed the phone getting laggy"

Unbounded memory growth is bad times.  @collinjackson @Hixie

### PR DESCRIPTION
**Pull request by abarth**
Fixed #1009

### CODE REVIEW

**Comment by abarth** [comment]
@eseidelGoogle @Hixie 

I think the biggest controversy here is whether to take a production dependency on Quiver.

**Comment by Hixie** [comment]
That's not controversial. We already depend on quiver (via vector_math).

File: packages/flutter/pubspec.yaml [hunk 1]
```diff
@@ -11,6 +11,7 @@ dependencies:
   sky_engine: 0.0.75
   sky_services: 0.0.75
   vector_math: '>=1.4.3 <2.0.0'
+  quiver: '>=0.21.4 <1.0.0'
```
Comment (devoncarew) [inline packages/flutter/pubspec.yaml]: This should probably be `'>=0.21.4 <0.22.0', or just `^0.21.0`.

**Comment by Hixie** [comment]
lgtm

**Comment by abarth** [comment]
> That's not controversial. We already depend on quiver (via vector_math).

`pub deps` in packages/flutter says that vector_math doesn't have any dependencies.

**Comment by Hixie** [comment]
I see:
```
|
'-- vector_math 1.4.4
    '-- quiver 0.21.4
        '-- matcher...
```

Might be a dev dependency?

**Comment by Hixie** [comment]
https://github.com/google/vector_math.dart/blob/master/pubspec.yaml says it's a regular dependency

**Comment by abarth** [comment]
vector_math 1.4.3

Maybe a recent change.

**Comment by abarth** [comment]
Ok, will land.

**Comment by Hixie** [comment]
Ah, yeah, John release 1.4.4 a couple of weeks ago. The dependency comes from a chance last summer, but it only just got released.

**Pull request merged**